### PR TITLE
Issue/9284 after login redirection

### LINKED ISSUE #9284: App redirects back to MyStore when user clicks on the launcher icon
**Describe the bug**
Within the same login session, the app will always open My Store screen, once the app was minimized and then opened again, no matter which screen was open before minimizing:

**To Reproduce**
1. Login into the app (if you are logged in already, log out and log in)!!
2. Start creating a new product and add for example title
3. Put the app into background
4. Tap on the launcher icon of the app to bring it back to foreground
5. Notice the app opens My Store screen and the product creation data are lost

**Expected behavior**
The app should open the order creation in the same state the user left it.

### PR DESCRIPTION
**Pull request by malinajirka**
Closes: #9284 

### Description

This PR updates the launchMode of MainActivity to ensure a new instance is not started when the activity is already at the top of the stack. 

This PR doesn't implement an optimal fix as it'd be quite complex and is imo not worth the time at the moment. It seems to me the app misuses the`FLAG_ACTIVITY_NEW_TASK` -> at least I don't see any reason why we'd need this flag in our app.

The ideal fix imo would be to remove the usage of `FLAG_ACTIVITY_NEW_TASK` from all the places and rely on the default Android behavior + ensure MainActivity isn't recreated when it's already at the top of the stack.

Either case, this PR only updates the latter part of the code and ensures the MainActivity is not recreated when it's already at the top. This fixes the original issue where the MainActivity used to redirect the user to MyStore screen when it was originally not launched from the OS Launcher but for example from the Login Flow.

More info on peaMlT-64-p2

Let me know if you think this fix is worth it - seems a bit risky, but not too risky, wdyt?

### Testing instructions

- Test Logout
- Test Login
- Test New Order Notification
- Test Shortcuts

1. Login into the app (if you are logged in already, log out and log in)!!
2. Start creating a new product and add for example title
3. Put the app into background
4. Tap on the launcher icon of the app to bring it back to foreground
5. Notice the state of Product Creation is retained

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -41,7 +41,8 @@
             android:name=".ui.main.MainActivity"
             android:exported="true"
             android:theme="@style/Theme.Woo.Splash"
-            android:windowSoftInputMode="adjustResize">
+            android:windowSoftInputMode="adjustResize"
+            android:launchMode="singleTop">
```
Comment (kidinov) [inline WooCommerce/src/main/AndroidManifest.xml:45]: Do not really understand what's going on, but even without this change, we have issues with the universal link. Apparently, It opens a second task. I think that when I was working on [this revision](https://github.com/woocommerce/woocommerce-android/pull/8466) it was working fine back then

```
adb shell am start -a android.intent.action.VIEW \
   -c android.intent.category.BROWSABLE \
   -d "https://woocommerce.com/mobile/payments/tap-to-pay"
```

https://github.com/woocommerce/woocommerce-android/assets/4923871/76bf2639-720c-4767-836d-7bbb3f0f81be

Comment (kidinov) [reply WooCommerce/src/main/AndroidManifest.xml:45]: Shortcuts work fine, while it is supposed to be the same as the command above, so maybe it's not 🤔

Maybe the issue is the way how the adb starts activity

**Review by kidinov — APPROVED** [review]
Thanks for working on that. We probably may merge that because it seems that even before the change, something not ok either with ADB or universal link handling/our strategy to handle tasks

File: WooCommerce/src/main/AndroidManifest.xml (hunk 1 above)
Comment (malinajirka) [reply WooCommerce/src/main/AndroidManifest.xml:45]: We discussed this in DM and it turns out this behavior is not present on the branch of this PR, so we are good to go :).

**Pull request merged**